File: src/main/java/com/vjiki/music/mapper/MessageMapper.java
package com.vjiki.music.mapper;

import com.vjiki.music.dto.MessageResponse;
import com.vjiki.music.entity.Message;

public final class MessageMapper {

    private MessageMapper() {
    }

    public static MessageResponse toResponse(Message message) {
        return MessageResponse.builder()
                .id(message.getId())
                .chatId(message.getChatId())
                .senderId(message.getSenderId())
                .senderEmail(message.getSender() == null ? null : message.getSender().getEmail())
                .senderNickname(message.getSender() == null ? null : message.getSender().getNickname())
                .senderAvatarUrl(message.getSender() == null ? null : message.getSender().getAvatarUrl())
                .replyToId(message.getReplyToId())
                .messageType(message.getMessageType() == null ? null : message.getMessageType().name())
                .content(message.getContent())
                .songId(message.getSongId())
                .attachmentCount(message.getAttachmentCount())
                .isEdited(message.getIsEdited())
                .isDeleted(message.getIsDeleted())
                .createdAt(message.getCreatedAt())
                .updatedAt(message.getUpdatedAt())
                .build();
    }
}

File: src/main/kotlin/com/vjiki/music/mapper/MessageMapper.kt
package com.vjiki.music.mapper

import com.vjiki.music.dto.MessageResponse
import com.vjiki.music.entity.Message
import org.springframework.stereotype.Component

@Component
object MessageMapper {
    fun Message.toResponse(): MessageResponse {
        return MessageResponse(
            id = id,
            chatId = chatId,
            senderId = senderId,
            senderEmail = sender?.email,
            senderNickname = sender?.nickname,
            senderAvatarUrl = sender?.avatarUrl,
            replyToId = replyToId,
            messageType = messageType.name,
            content = content,
            songId = songId,
            attachmentCount = attachmentCount,
            isEdited = isEdited,
            isDeleted = isDeleted,
            createdAt = createdAt,
            updatedAt = updatedAt
        )
    }
}

